qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,810 @@
+    _envelope_lam = LambdifiedExpression("envelope")
+    _constraints_lam = LambdifiedExpression("constraints")
+    _eval_conditions_lam = LambdifiedExpression("eval_conditions")
+
+    def __init__(
+        self,
+        pulse_type: str,
+        duration: Union[ParameterExpression, int],
+        parameters: Optional[Dict[str, Union[ParameterExpression, complex]]] = None,
+        name: Optional[str] = None,
+        limit_amplitude: Optional[bool] = None,
+        envelope: Optional[sym.Expr] = None,
+        constraints: Optional[sym.Expr] = None,
+        eval_conditions: Optional[sym.Expr] = None,
+    ):
+        """Create a parametric pulse.
+
+        Args:
+            pulse_type: Display name of this pulse shape.
+            duration: Duration of pulse.
+            parameters: Dictionary of pulse parameters that defines the pulse envelope.
+            name: Display name for this particular pulse envelope.
+            limit_amplitude: If ``True``, then limit the absolute value of the amplitude of the
+                waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+            envelope: Pulse envelope expression.
+            constraints: Pulse parameter constraint expression.
+            eval_conditions: Extra conditions to evaluate a full waveform to check the
+                amplitude limit. If this condition is met, then the validation routine
+                will investigate the full waveform and raise an error when the amplitude norm
+                of any data point exceeds 1.0. If not provided, the validation always
+                creates a full waveform.
+
+        Raises:
+            PulseError: When not all parameters are listed in the attribute :attr:`PARAM_DEF`.
+        """
+        super().__init__(
+            duration=duration,
+            name=name,
+            limit_amplitude=limit_amplitude,
+        )
+        if parameters is None:
+            parameters = {}
+
+        self._pulse_type = pulse_type
+        self._params = parameters
+
+        self.envelope = envelope
+        self.constraints = constraints
+        self.eval_conditions = eval_conditions
+
+    def __getattr__(self, item):
+        # Get pulse parameters with attribute-like access.
+        params = object.__getattribute__(self, "_params")
+        if item not in params:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{item}'")
+        return params[item]
+
+    @property
+    def pulse_type(self) -> str:
+        """Return display name of the pulse shape."""
+        return self._pulse_type
+
+    def get_waveform(self) -> Waveform:
+        r"""Return a Waveform with samples filled according to the formula that the pulse
+        represents and the parameter values it contains.
+
+        Since the returned array is a discretized time series of the continuous function,
+        this method uses a midpoint sampler. For ``duration``, return:
```

Comment:
Should the sampler be an option that the user can provide? If it is not provided we default to `midpoint`.

Reply:
I think this is overengineering for now. Indeed `ParametricPulse` doesn't have such flexibility. We can add this later since this upgrade doesn't break any public API.